ansible/roles/test/files/saitests/sai_qos_tests.py

```diff
@@ -736,160 +799,191 @@ def runTest(self):
+        print >> sys.stderr, self.src_port_ips
         sys.stderr.flush()
 
-        dst_port_id = self.test_params['dst_port_id']
-        dst_port_ip = self.test_params['dst_port_ip']
-        pgs_num = self.test_params['pgs_num']
-        asic_type = self.test_params['sonic_asic_type']
-        pkts_num_leak_out = self.test_params['pkts_num_leak_out']
-        pkts_num_trig_pfc = self.test_params['pkts_num_trig_pfc']
-        pkts_num_hdrm_full = self.test_params['pkts_num_hdrm_full']
-        pkts_num_hdrm_partial = self.test_params['pkts_num_hdrm_partial']
-        print >> sys.stderr, ("pkts num: leak_out: %d, trig_pfc: %d, hdrm_full: %d, hdrm_partial: %d" % (pkts_num_leak_out, pkts_num_trig_pfc, pkts_num_hdrm_full, pkts_num_hdrm_partial))
+        self.dst_port_id = self.test_params['dst_port_id']
+        self.dst_port_ip = self.test_params['dst_port_ip']
+        self.pgs_num = self.test_params['pgs_num']
+        self.asic_type = self.test_params['sonic_asic_type']
+        self.pkts_num_leak_out = self.test_params['pkts_num_leak_out']
+        self.pkts_num_trig_pfc = self.test_params['pkts_num_trig_pfc']
+        self.pkts_num_hdrm_full = self.test_params['pkts_num_hdrm_full']
+        self.pkts_num_hdrm_partial = self.test_params['pkts_num_hdrm_partial']
+        print >> sys.stderr, ("pkts num: leak_out: %d, trig_pfc: %d, hdrm_full: %d, hdrm_partial: %d" % (self.pkts_num_leak_out, self.pkts_num_trig_pfc, self.pkts_num_hdrm_full, self.pkts_num_hdrm_partial))
         sys.stderr.flush()
 
-        dst_port_mac = self.dataplane.get_mac(0, dst_port_id)
-        src_port_macs = [self.dataplane.get_mac(0, ptid) for ptid in src_port_ids]
+        self.dst_port_mac = self.dataplane.get_mac(0, self.dst_port_id)
+        self.src_port_macs = [self.dataplane.get_mac(0, ptid) for ptid in self.src_port_ids]
+
+        if self.testbed_type in ['t0', 't0-64', 't0-116']:
+            # populate ARP
+            for idx, ptid in enumerate(self.src_port_ids):
+
+                arpreq_pkt = simple_arp_packet(
+                              eth_dst='ff:ff:ff:ff:ff:ff',
+                              eth_src=self.src_port_macs[idx],
+                              arp_op=1,
+                              ip_snd=self.src_port_ips[idx],
+                              ip_tgt='192.168.0.1',
+                              hw_snd=self.src_port_macs[idx],
+                              hw_tgt='00:00:00:00:00:00')
+                send_packet(self, ptid, arpreq_pkt)
+            arpreq_pkt = simple_arp_packet(
+                          eth_dst='ff:ff:ff:ff:ff:ff',
+                          eth_src=self.dst_port_mac,
+                          arp_op=1,
+                          ip_snd=self.dst_port_ip,
+                          ip_tgt='192.168.0.1',
+                          hw_snd=self.dst_port_mac,
+                          hw_tgt='00:00:00:00:00:00')
+            send_packet(self, self.dst_port_id, arpreq_pkt)
+        time.sleep(8)
```

Comment:
Can this duplicate code handled by class ARPpopulate()?

Reply:
I would need to pass in a lot of variables to the ARPpopulate if i need it to handle this one by calling ARPpopulate from this test. 
The other option is to pass in all the variables used by this test for arp population along with the existing set and modify the ARPpopulate class to handle everything.
I prefer the second one if this change is needed.